File: DoItTest.Domain/Tests/TestItems/AnswerOptions/CheckboxesAnswerOption.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DoItTest.Domain.Tests.TestItems.AnswerOptions
{
    public class CheckboxesAnswerOption: AnswerOption
    {
        public String Title { get; }
        public Boolean IsTrue { get; }

        public CheckboxesAnswerOption(Guid id, Guid testItemId, TestItemType type, String title, Boolean isTrue) : base(id, testItemId, type)
        {
            Title = title;
            IsTrue = isTrue;
        }
    }
}

File: DoItTest/ClientApp/src/domain/tests/items/answerOption/checkboxesAnswerOption.ts
import { TestItemType } from "../testItemType"
import { AnswerOption } from "./answerOption"

export class CheckboxesAnswerOption extends AnswerOption {
    constructor(
        public id: string,
        public testItemId: string,
        public type: TestItemType,
        public title: string,
        public isTrue: boolean | null
    ) {
        super(id, testItemId, type)
    }
}

export function mapCheckboxesAnswerOption(value: any): CheckboxesAnswerOption {
    return new CheckboxesAnswerOption(value.id, value.testItemId, value.type, value.title, value.isTrue)
}

export function mapCheckboxesAnswerOptions(value: any[]): CheckboxesAnswerOption[] {
    return value.map(mapCheckboxesAnswerOption)
}